### PR TITLE
Feature/db handler rework

### PR DESCRIPTION
**Pull request by Zelle97**
Implemented Database migration with knex.js and rewrote DatabaseHandler to async/await.

### CODE REVIEW

**Comment by Zelle97** [comment]
Also check the changes in the weebs-catalog to see what changed with the docker-compose and database stuff.

File: handler/util/mariadbHandler.js [hunk 1]
```diff
@@ -16,117 +16,59 @@ pool.on('release', function(connection) {
...
-                    resolve(results);
-                    connection.release();
-                });
-            });
-        });
+    getFavPlaylist: async function(userId) {
+        const connection = await pool.getConnection();
+        const getFavPlaylist = `SELECT favPlaylist FROM discordUser WHERE discordId = ${userId}`;
+        try {
+            // TODO Possible checks
```
Comment (Emdix) [inline handler/util/mariadbHandler.js]: possible checks?

File: knexfile.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const secret = require('./configuration/secrets');
```
Comment (Emdix) [inline knexfile.js]: why not use the secretHandler file?

File: knexfile.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+module.exports = {
+    client: 'mysql',
+    connection: {
```
Comment (Emdix) [inline knexfile.js]: maybe use 
``connection = secretHandler.getDatabaseSecrets();``
for simplicity?

File: handler/util/mariadbHandler.js [hunk 4]
```diff
@@ -16,117 +16,59 @@ pool.on('release', function(connection) {
...
-                    resolve(results);
-                });
-                connection.release();
-            });
-        });
+    getGuildPrefix: async function(guildId) {
+        const connection = await pool.getConnection();
+        const getGuildPrefix = `SELECT prefix FROM guildConfiguration WHERE guildId = ${guildId}`;
+        try {
+            // TODO possible checks
```
Comment (Emdix) [inline handler/util/mariadbHandler.js]: possible checks?

**Pull request merged**